### PR TITLE
Skip permission updates for Homebrew folders when NOROOT is set

### PR DESCRIPTION
**Pull request by isnotgood**

### CODE REVIEW

File: tools/setup-buildtools-apple.sh
Comment (lalitb) [inline tools/setup-buildtools-apple.sh:18]: Should we add a caution that the build may fail if another user takes ownership of these files?

**Review by lalitb — APPROVED** [review]
LGTM other than a minor comment.

File: tools/setup-buildtools-apple.sh
Comment (isnotgood) [reply tools/setup-buildtools-apple.sh]: Good point, pls take a look at new commit. I think I improved the error reporting. If you think it's unnecessary change I'll revert and just extend the origin error message.

Comment (lalitb) [reply tools/setup-buildtools-apple.sh]: Thanks. This looks good now. Not related to this PR, but looks like CI build is failing for Linux and Android, PR can be merged once that is fixed. Looking into it now.

Comment (lalitb) [reply tools/setup-buildtools-apple.sh]: Can you now try to rebase the PR branch with main, to validate if the CI build is successful?

Comment (isnotgood) [reply tools/setup-buildtools-apple.sh]: Done, let's see how CI will go :)

Comment (lalitb) [reply tools/setup-buildtools-apple.sh]: Thanks. Should be good to merge. CodeQL scan is failing, #967 will address this.

Comment (lalitb) [reply tools/setup-buildtools-apple.sh]: Should be good to merge.  The CodeQL failure can be ignored for now, we have issue #967 for that.

Comment (isnotgood) [reply tools/setup-buildtools-apple.sh]: cool, can you pls merge it (I don't have write access, just a visitor here :)) or we're waiting for more reviewers?

**Pull request merged**
